### PR TITLE
Add typos to CI

### PR DESCRIPTION
**Pull request by jayvdb**
Please merge https://github.com/juhaku/utoipa/pull/1038 first

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -87,6 +91,17 @@ jobs:
           if [[ ${{ steps.changes.outputs.changes }} == true ]]; then
             ./scripts/test.sh ${{ matrix.crate }}
           fi
+
+  check-typos:
+    name: typos
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (juhaku) [inline .github/workflows/build.yaml:99]: What does this do? Does it throw error if it finds typos?

Comment (jayvdb) [reply .github/workflows/build.yaml:99]: yup , built with rust https://crates.io/crates/typos-cli

c.f. https://github.com/taiki-e/install-action

Comment (jayvdb) [reply .github/workflows/build.yaml:99]: it is configurable if there are false positives

Comment (juhaku) [reply .github/workflows/build.yaml:99]: Alright, I guess we can have it.

**Pull request merged**